File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        std::vector<int> result{};
        if (root == nullptr) return result;

        std::queue<TreeNode*> my_queue{};
        my_queue.push(root);
        while (!my_queue.empty()) {
            int size = my_queue.size();
            for (int i = 0; i < size; i++) {
                TreeNode* node = my_queue.front();
                my_queue.pop();
                if (node->left != nullptr) my_queue.push(node->left);
                if (node->right != nullptr) my_queue.push(node->right);

                if (i == size - 1) result.push_back(node->val);
            }
        }

        return result;
    }
};

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {    
    public bool IsSubtree(TreeNode root, TreeNode subRoot) {
        string rootString = SerializeTree(root);
        string subRootString = SerializeTree(subRoot);

        return rootString.Contains(subRootString);
    }

    private string SerializeTree(TreeNode root) {
        if (root == null) return "#";

        return root.val + "," + SerializeTree(root.left) + "," + SerializeTree(root.right);
    }

}
